src/System.Private.Interop/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -111,6 +111,44 @@ public static String PtrToStringAuto(IntPtr ptr)
             return PtrToStringUni(ptr);
         }
 
+        public static unsafe String PtrToStringUTF8(IntPtr ptr)
+        {
+            if (IntPtr.Zero == ptr)
+            {
+                return null;
+            }
+            else
+            {
+                int nbBytes = lstrlenA(ptr);
+                return PtrToStringUTF8(ptr, nbBytes);
+            }
+        }
+
+        public static unsafe String PtrToStringUTF8(IntPtr ptr, int byteLen)
+        {
+            if (byteLen < 0)
+            {
+                throw new ArgumentException(null, nameof(byteLen));
```

Comment:
This should probably have a message string

Reply:
(Apparently CoreCLR has this bug too, but nevertheless, both should have a message.)